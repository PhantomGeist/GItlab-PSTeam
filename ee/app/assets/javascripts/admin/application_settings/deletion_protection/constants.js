import { __, s__ } from '~/locale';

export const I18N_DELETION_PROTECTION = {
  label: s__('DeletionSettings|Deletion protection'),
  helpText: s__(
    'DeletionSettings|Period that deleted groups and projects will remain restorable for. Personal projects are always deleted immediately.',
  ),
  learnMore: __('Learn more.'),
  days: __('days'),
};
