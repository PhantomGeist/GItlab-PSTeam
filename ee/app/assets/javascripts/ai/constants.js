import { s__, sprintf } from '~/locale';

export const i18n = {
  GITLAB_DUO: s__('AI|GitLab Duo'),
  GENIE_TOOLTIP: s__('AI|What does the selected code mean?'),
  GENIE_NO_CONTAINER_ERROR: s__("AI|The container element wasn't found, stopping AI Genie."),
  GENIE_CHAT_TITLE: s__('AI|Code Explanation'),
  GENIE_CHAT_CLOSE_LABEL: s__('AI|Close the Code Explanation'),
  GENIE_CHAT_LEGAL_NOTICE: sprintf(
    s__(
      'AI|You are not allowed to copy any part of this output into issues, comments, GitLab source code, commit messages, merge requests or any other user interface in the %{gitlabOrg} or %{gitlabCom} groups.',
    ),
    { gitlabOrg: '<code>/gitlab-org</code>', gitlabCom: '<code>/gitlab-com</code>' },
    false,
  ),
  GENIE_CHAT_LEGAL_GENERATED_BY_AI: s__('AI|Responses generated by AI'),
  REQUEST_ERROR: s__('AI|Something went wrong. Please try again later'),
  EXPERIMENT_BADGE: s__('AI|Experiment'),
  EXPERIMENT_POPOVER_TITLE: s__("AI|What's an Experiment?"),
  EXPERIMENT_POPOVER_CONTENT: s__(
    "AI|An %{linkStart}Experiment%{linkEnd} is a feature that's in the process of being developed. It's not production-ready. We encourage users to try Experimental features and provide feedback. An Experiment: %{bullets}",
  ),
  EXPERIMENT_POPOVER_BULLETS: [
    s__('AI|May be unstable'),
    s__('AI|Has no support and might not be documented'),
    s__('AI|Can be removed at any time'),
  ],
  EXPLAIN_CODE_PROMPT: s__(
    'AI|Explain the code from %{filePath} in human understandable language presented in Markdown format. In the response add neither original code snippet nor any title. `%{text}`. If it is not programming code, say `The selected text is not code. I am afraid this feature is for explaining code only. Would you like to ask a different question about the selected text?` and wait for another question.',
  ),
  TOO_LONG_ERROR_MESSAGE: s__(
    'AI|There is too much text in the chat. Please try again with a shorter text.',
  ),
  GENIE_CHAT_PROMPT_PLACEHOLDER: s__('AI|GitLab Duo Chat'),
  GENIE_CHAT_SUBMIT_LABEL: s__('AI|Send chat message.'),
  GENIE_CHAT_EMPTY_STATE_TITLE: s__('AI|Ask a question'),
  GENIE_CHAT_EMPTY_STATE_DESC: s__('AI|AI generated explanations will appear here.'),
  GENIE_CHAT_LEGAL_DISCLAIMER: s__(
    "AI|May provide inappropriate responses not representative of GitLab's views. Do not input personal data.",
  ),
  GENIE_CHAT_NEW_CHAT: s__('AI|New chat'),
  GENIE_CHAT_LOADING_MESSAGE: s__('AI|%{tool} is %{transition} an answer'),
  GENIE_CHAT_LOADING_TRANSITIONS: [
    s__('AI|finding'),
    s__('AI|working on'),
    s__('AI|generating'),
    s__('AI|producing'),
  ],
  GENIE_CHAT_FEEDBACK_LINK: s__('AI|Give feedback to improve this answer.'),
  GENIE_CHAT_FEEDBACK_THANKS: s__('AI|Thank you for your feedback.'),
};
export const GENIE_CHAT_LOADING_TRANSITION_DURATION = 7500;
export const TOO_LONG_ERROR_TYPE = 'too-long';
export const AI_GENIE_DEBOUNCE = 300;
export const GENIE_CHAT_MODEL_ROLES = {
  user: 'user',
  system: 'system',
  assistant: 'assistant',
};

export const CHAT_MESSAGE_TYPES = {
  tool: 'tool',
};

export const FEEDBACK_OPTIONS = [
  {
    title: s__('AI|Helpful'),
    icon: 'thumb-up',
    value: 'helpful',
  },
  {
    title: s__('AI|Unhelpful'),
    icon: 'thumb-down',
    value: 'unhelpful',
  },
  {
    title: s__('AI|Wrong'),
    icon: 'status_warning',
    value: 'wrong',
  },
];

export const EXPLAIN_CODE_TRACKING_EVENT_NAME = 'explain_code_blob_viewer';
export const TANUKI_BOT_TRACKING_EVENT_NAME = 'ask_gitlab_chat';
export const GENIE_CHAT_RESET_MESSAGE = '/reset';

export const DOCUMENTATION_SOURCE_TYPES = {
  HANDBOOK: {
    value: 'handbook',
    icon: 'book',
  },
  DOC: {
    value: 'doc',
    icon: 'documents',
  },
  BLOG: {
    value: 'blog',
    icon: 'list-bulleted',
  },
};
