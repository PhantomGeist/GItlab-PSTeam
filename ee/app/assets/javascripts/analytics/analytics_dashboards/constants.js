import { s__ } from '~/locale';

export const FEATURE_PRODUCT_ANALYTICS = 'productAnalytics';

export const PRODUCT_ANALYTICS_FEATURE_DASHBOARDS = ['audience', 'behavior'];

export const EVENTS_TYPES = ['pageViews', 'linkClickEvents', 'events'];

export function isTrackedEvent(eventType) {
  return EVENTS_TYPES.includes(eventType);
}

export const PANEL_VISUALIZATION_HEIGHT = '600px';

export const PANEL_DISPLAY_TYPES = {
  DATA: 'data',
  VISUALIZATION: 'visualization',
  CODE: 'code',
};

export const PANEL_DISPLAY_TYPE_ITEMS = [
  {
    type: PANEL_DISPLAY_TYPES.DATA,
    icon: 'table',
    title: s__('Analytics|Data'),
  },
  {
    type: PANEL_DISPLAY_TYPES.VISUALIZATION,
    icon: 'chart',
    title: s__('Analytics|Visualization'),
  },
  {
    type: PANEL_DISPLAY_TYPES.CODE,
    icon: 'code',
    title: s__('Analytics|Code'),
  },
];

export const MEASURE_COLOR = '#00b140';
export const DIMENSION_COLOR = '#c3e6cd';

export const EVENTS_TABLE_NAME = 'TrackedEvents';
export const SESSIONS_TABLE_NAME = 'Sessions';

export const ANALYTICS_FIELD_CATEGORIES = [
  {
    name: s__('Analytics|Pages'),
    category: 'pages',
  },
  {
    name: s__('Analytics|Users'),
    category: 'users',
  },
  {
    name: s__('Analytics|Link clicks'),
    category: 'linkClicks',
  },
  {
    name: s__('Analytics|Custom events'),
    category: 'custom',
  },
];

export const ANALYTICS_FIELDS = [
  {
    name: s__('Analytics|URL'),
    category: 'pages',
    dbField: 'pageUrl',
    icon: 'documents',
  },
  {
    name: s__('Analytics|Page Path'),
    category: 'pages',
    dbField: 'pageUrlpath',
    icon: 'documents',
  },
  {
    name: s__('Analytics|Page Title'),
    category: 'pages',
    dbField: 'pageTitle',
    icon: 'documents',
  },
  {
    name: s__('Analytics|Page Language'),
    category: 'pages',
    dbField: 'documentLanguage',
    icon: 'documents',
  },
  {
    name: s__('Analytics|Host'),
    category: 'pages',
    dbField: 'pageUrlhosts',
    icon: 'documents',
  },
  {
    name: s__('Analytics|Referer'),
    category: 'users',
    dbField: 'pageReferrer',
    icon: 'user',
  },
  {
    name: s__('Analytics|Language'),
    category: 'users',
    dbField: 'browserLanguage',
    icon: 'user',
  },
  {
    name: s__('Analytics|Viewport'),
    category: 'users',
    dbField: 'viewportSize',
    icon: 'user',
  },
  {
    name: s__('Analytics|Browser Family'),
    category: 'users',
    dbField: 'agentName',
    icon: 'user',
  },
  {
    name: s__('Analytics|Browser'),
    category: 'users',
    dbField: ['agentName', 'agentVersion'],
    icon: 'user',
  },
  {
    name: s__('Analytics|OS'),
    category: 'users',
    dbField: 'osName',
    icon: 'user',
  },
  {
    name: s__('Analytics|OS Version'),
    category: 'users',
    dbField: ['osName', 'osVersion'],
    icon: 'user',
  },
  {
    name: s__('Analytics|User Id'),
    category: 'users',
    dbField: 'userId',
    icon: 'user',
  },
  {
    name: s__('Analytics|Target URL'),
    category: 'linkClicks',
    dbField: ['targetUrl'],
    icon: 'link',
  },
  {
    name: s__('Analytics|Element ID'),
    category: 'linkClicks',
    dbField: ['elementId'],
    icon: 'link',
  },
  {
    name: s__('Analytics|Event Name'),
    category: 'custom',
    dbField: 'customEventName',
    icon: 'documents',
  },
  {
    name: s__('Analytics|Event Props'),
    category: 'custom',
    dbField: 'customEventProps',
    icon: 'documents',
  },
  {
    name: s__('Analytics|User Props'),
    category: 'custom',
    dbField: 'customUserProps',
    icon: 'user',
  },
];

export const NEW_DASHBOARD = () => ({
  title: '',
  description: '',
  panels: [],
  userDefined: true,
});

export const FILE_ALREADY_EXISTS_SERVER_RESPONSE = 'A file with this name already exists';
