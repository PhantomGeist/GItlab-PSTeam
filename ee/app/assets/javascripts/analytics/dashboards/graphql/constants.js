export const BUCKETING_INTERVAL_ALL = 'ALL';

export const MERGE_REQUESTS_STATE_MERGED = 'merged';
