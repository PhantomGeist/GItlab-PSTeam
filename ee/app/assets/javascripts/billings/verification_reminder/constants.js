import { s__ } from '~/locale';
import { DOCS_URL } from 'jh_else_ce/lib/utils/url_utility';

export const FEATURE_NAME = 'verification_reminder';
export const DOCS_LINK = `${DOCS_URL}/runner/install/`;
export const EVENT_LABEL = 'verification_reminder';
export const MOUNTED_EVENT = 'shown';
export const DISMISS_EVENT = 'dismissed';
export const OPEN_DOCS_EVENT = 'clicked_docs_link';
export const START_VERIFICATION_EVENT = 'start_verification';
export const SUCCESSFUL_VERIFICATION_EVENT = 'successful_verification';
export const I18N = {
  warningAlert: {
    title: s__(
      'VerificationReminder|Pipeline failing? To keep GitLab spam and abuse free we ask that you verify your identity.',
    ),
    message: s__(
      'VerificationReminder|Until then, shared runners will be unavailable. %{validateLinkStart}Validate your account%{validateLinkEnd} or %{docsLinkStart}use your own runners%{docsLinkEnd}.',
    ),
  },
  successAlert: {
    title: s__('VerificationReminder|Your account has been validated'),
    message: s__(
      'VerificationReminder|Youâ€™ll now be able to take advantage of free compute minutes on shared runners.',
    ),
  },
};
