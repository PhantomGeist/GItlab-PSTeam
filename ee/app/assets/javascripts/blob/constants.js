import { n__, s__ } from '~/locale';
import { DOCS_URL_IN_EE_DIR } from 'jh_else_ce/lib/utils/url_utility';

export const SYNTAX_ERRORS_TEXT = (count) =>
  n__(
    'CodeownersValidation|Contains %d syntax error.',
    'CodeownersValidation|Contains %d syntax errors.',
    count,
  );

export const SYNTAX_VALID = s__('CodeownersValidation|Syntax is valid.');

export const DOCS_LINK_TEXT = s__('CodeownersValidation|How are errors handled?');

export const SHOW_ERRORS = s__('CodeownersValidation|Show errors');

export const HIDE_ERRORS = s__('CodeownersValidation|Hide errors');

export const COLLAPSE_ID = 'CODEOWNERS_VALIDATION_COLLAPSE';

export const LINE = s__('CodeownersValidation|Line');

export const ERROR_MESSAGE = s__(
  'CodeownersValidation|An error occurred while loading the validation errors. Please try again later.',
);

export const DOCS_URL = `${DOCS_URL_IN_EE_DIR}/user/project/codeowners/reference.html#error-handling-in-code-owners`;

export const CODEOWNERS_VALIDATION_I18N = {
  syntaxValid: SYNTAX_VALID,
  syntaxErrors: SYNTAX_ERRORS_TEXT,
  show: SHOW_ERRORS,
  hide: HIDE_ERRORS,
  docsLink: DOCS_LINK_TEXT,
  line: LINE,
  errorMessage: ERROR_MESSAGE,
};

export const INVALID_SECTION_OWNER_FORMAT = s__('CodeownersValidation|Inaccessible owners');

export const MISSING_ENTRY_OWNER = s__('CodeownersValidation|Zero owners');

export const INVALID_ENTRY_OWNER_FORMAT = s__('CodeownersValidation|Entries with spaces');

export const MISSING_SECTION_NAME = s__('CodeownersValidation|Missing section name');

export const INVALID_APPROVAL_REQUIREMENT = s__(
  'CodeownersValidation|Less than 1 required approvals',
);

export const INVALID_SECTION_FORMAT = s__('CodeownersValidation|Unparsable sections');

export const CODE_TO_MESSAGE = {
  invalid_section_owner_format: INVALID_SECTION_OWNER_FORMAT,
  missing_entry_owner: MISSING_ENTRY_OWNER,
  invalid_entry_owner_format: INVALID_ENTRY_OWNER_FORMAT,
  missing_section_name: MISSING_SECTION_NAME,
  invalid_approval_requirement: INVALID_APPROVAL_REQUIREMENT,
  invalid_section_format: INVALID_SECTION_FORMAT,
};
