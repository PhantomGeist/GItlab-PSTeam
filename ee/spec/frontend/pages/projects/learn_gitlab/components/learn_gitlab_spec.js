import { GlProgressBar, GlAlert } from '@gitlab/ui';
import { mount } from '@vue/test-utils';
import Vue from 'vue';
import Cookies from '~/lib/utils/cookies';
import LearnGitlab from 'ee/pages/projects/learn_gitlab/components/learn_gitlab.vue';
import UltimateTrialBenefitModal from 'ee/pages/projects/learn_gitlab/components/ultimate_trial_benefit_modal.vue';
import eventHub from '~/invite_members/event_hub';
import { INVITE_MODAL_OPEN_COOKIE } from 'ee/pages/projects/learn_gitlab/constants';
import { ON_CELEBRATION_TRACK_LABEL } from '~/invite_members/constants';
import eventHubNav from '~/super_sidebar/event_hub';
import { stubComponent } from 'helpers/stub_component';
import {
  testActions,
  testSections,
  testProject,
  testProjectShowUltimateTrialBenefitModal,
} from './mock_data';

Vue.config.ignoredElements = ['gl-emoji'];

describe('Learn GitLab', () => {
  let wrapper;
  let sidebar;
  let showModal;

  const createWrapper = ({ project = testProject } = {}) => {
    wrapper = mount(LearnGitlab, {
      propsData: {
        actions: testActions,
        sections: testSections,
        project,
      },
      stubs: {
        UltimateTrialBenefitModal: stubComponent(UltimateTrialBenefitModal, {
          methods: {
            show: showModal,
          },
        }),
      },
    });
  };

  describe('Initial rendering concerns', () => {
    beforeEach(() => {
      showModal = jest.fn();
      createWrapper();
    });

    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders the progress percentage', () => {
      const text = wrapper.find('[data-testid="completion-percentage"]').text();

      expect(text).toBe('25% completed');
    });

    it('renders the progress bar with correct values', () => {
      const progressBar = wrapper.findComponent(GlProgressBar);

      expect(progressBar.vm.$attrs.value).toBe(3);
      expect(progressBar.vm.$attrs.max).toBe(12);
    });
  });

  describe('Invite Members Modal', () => {
    let spy;
    let cookieSpy;

    beforeEach(() => {
      spy = jest.spyOn(eventHub, '$emit');
      cookieSpy = jest.spyOn(Cookies, 'remove');
    });

    afterEach(() => {
      Cookies.remove(INVITE_MODAL_OPEN_COOKIE);
    });

    it('emits openModal', () => {
      Cookies.set(INVITE_MODAL_OPEN_COOKIE, true);

      createWrapper();

      expect(spy).toHaveBeenCalledWith('openModal', {
        mode: 'celebrate',
        source: ON_CELEBRATION_TRACK_LABEL,
      });
      expect(cookieSpy).toHaveBeenCalledWith(INVITE_MODAL_OPEN_COOKIE);
    });

    it('does not emit openModal when cookie is not set', () => {
      createWrapper();

      expect(spy).not.toHaveBeenCalled();
      expect(cookieSpy).toHaveBeenCalledWith(INVITE_MODAL_OPEN_COOKIE);
    });
  });

  describe('Ultimate Trial Benefit Modal', () => {
    let cookieSpy;

    beforeEach(() => {
      cookieSpy = jest.spyOn(Cookies, 'remove');
    });

    it('calls show method when in experiment candidate with cookie set', async () => {
      Cookies.set(INVITE_MODAL_OPEN_COOKIE, true);

      await createWrapper({ project: testProjectShowUltimateTrialBenefitModal });

      expect(cookieSpy).toHaveBeenCalledWith(INVITE_MODAL_OPEN_COOKIE);
      expect(showModal).toHaveBeenCalled();
    });

    it('does not call show method in ultimate_trial_benefit_modal when in experiment candidate and cookie is not set', async () => {
      await createWrapper({ project: testProjectShowUltimateTrialBenefitModal });

      expect(cookieSpy).toHaveBeenCalledWith(INVITE_MODAL_OPEN_COOKIE);
      expect(showModal).not.toHaveBeenCalled();
    });
  });

  describe('when the showSuccessfulInvitationsAlert event is fired', () => {
    const findAlert = () => wrapper.findComponent(GlAlert);

    beforeEach(() => {
      createWrapper();
      eventHub.$emit('showSuccessfulInvitationsAlert');
    });

    it('displays the successful invitations alert', () => {
      expect(findAlert().exists()).toBe(true);
    });

    it('displays a message with the project name', () => {
      expect(findAlert().text()).toBe(
        "Your team is growing! You've successfully invited new team members to the test-project project.",
      );
    });
  });

  describe('with sidebar percentage updates', () => {
    let spy;

    beforeEach(() => {
      spy = jest.spyOn(eventHubNav, '$emit');
    });

    it('modifies the sidebar percentage', () => {
      sidebar = document.createElement('div');
      sidebar.innerHTML = `
    <div class="sidebar-top-level-items">
      <div class="active">
        <div class="count"></div>
      </div>
    </div>
    `;
      document.body.appendChild(sidebar);
      createWrapper();

      eventHub.$emit('showSuccessfulInvitationsAlert');

      expect(sidebar.textContent.trim()).toBe('25%');
      expect(spy).not.toHaveBeenCalled();

      sidebar.remove();
    });

    it('emits updatePillValue event for super sidebar', () => {
      createWrapper();

      eventHub.$emit('showSuccessfulInvitationsAlert');

      expect(spy).toHaveBeenCalledWith('updatePillValue', {
        value: '25%',
        itemId: 'learn_gitlab',
      });
    });
  });
});
