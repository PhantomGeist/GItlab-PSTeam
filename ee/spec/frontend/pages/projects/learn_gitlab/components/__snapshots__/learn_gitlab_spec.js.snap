// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Learn GitLab Initial rendering concerns renders correctly 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="gl-mb-7 gl-ml-5"
    >
      <h1
        class="gl-font-size-h1"
      >
        Learn GitLab
      </h1>
      <p
        class="gl-mb-0 gl-text-gray-700"
      >
        Ready to get started with GitLab? Follow these steps to set up your workspace, plan and commit changes, and deploy your project.
      </p>
    </div>
  </div>
  <div
    class="gl-mb-3"
  >
    <p
      class="gl-mb-2 gl-text-gray-500"
      data-testid="completion-percentage"
    >
      25% completed
    </p>
    <div
      class="progress"
    >
      <div
        aria-valuemax="12"
        aria-valuemin="0"
        aria-valuenow="3"
        class="progress-bar"
        role="progressbar"
        style="width: 25%;"
      />
    </div>
  </div>
  <div
    class="gl-mt-6"
  >
    <h2
      class="gl-font-size-h2 gl-font-weight-bold"
    >
      1. Add code to your project
    </h2>
  </div>
  <div
    class="row"
  >
    <div
      class="col-lg-3 col-mb-6 col-sm-12 gl-mt-5"
    >
      <div
        class="gl-card gl-pt-0 h-100"
      >
        <div
          class="gl-bg-white gl-border-0 gl-card-header gl-pb-0"
        >
          <img
            src="code.svg"
          />
          <h2
            class="gl-font-lg gl-mb-3"
          >
            Start with the WebIDE
          </h2>
          <p
            class="gl-mb-6 gl-text-gray-700"
          >
            Use the built-in editor to create or upload files.
          </p>
        </div>
        <div
          class="gl-card-body gl-pt-0"
        >
          <div
            class="gl-mb-4"
          >
            <div
              class="align-items-center flex"
            >
              <span
                class="gl-text-green-500"
              >
                <svg
                  aria-hidden="true"
                  class="gl-icon s16"
                  data-testid="completed-icon"
                  role="img"
                >
                  <use
                    href="file-mock#check-circle-filled"
                  />
                </svg>
                Add code
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="gl-mt-6"
  >
    <h2
      class="gl-font-size-h2 gl-font-weight-bold"
    >
      2. Build
    </h2>
  </div>
  <div
    class="row"
  >
    <div
      class="col-lg-4 col-mb-6 col-sm-12 gl-mt-5"
    >
      <div
        class="gl-card gl-pt-0 h-100"
      >
        <div
          class="gl-bg-white gl-border-0 gl-card-header gl-pb-0"
        >
          <img
            src="workspace.svg"
          />
          <h2
            class="gl-font-lg gl-mb-3"
          >
            Set up your workspace
          </h2>
          <p
            class="gl-mb-6 gl-text-gray-700"
          >
            Complete these tasks first so you can enjoy GitLab's features to their fullest:
          </p>
        </div>
        <div
          class="gl-card-body gl-pt-0"
        >
          <div
            class="gl-mb-4"
          >
            <div
              class="align-items-center flex"
            >
              <span
                class="gl-text-green-500"
              >
                <svg
                  aria-hidden="true"
                  class="gl-icon s16"
                  data-testid="completed-icon"
                  role="img"
                >
                  <use
                    href="file-mock#check-circle-filled"
                  />
                </svg>
                Invite your colleagues
              </span>
            </div>
          </div>
          <div
            class="gl-mb-4"
          >
            <div
              class="align-items-center flex"
            >
              <span
                class="gl-text-green-500"
              >
                <svg
                  aria-hidden="true"
                  class="gl-icon s16"
                  data-testid="completed-icon"
                  role="img"
                >
                  <use
                    href="file-mock#check-circle-filled"
                  />
                </svg>
                Create a repository
              </span>
            </div>
          </div>
          <div
            class="gl-mb-4"
          >
            <div
              class="align-items-center flex"
            >
              <div>
                <a
                  class="gl-link"
                  data-testid="uncompleted-learn-gitlab-link"
                  data-track-action="click_link"
                  data-track-label="set_up_your_first_project_s_ci_cd"
                  href="http://example.com/"
                  target="_self"
                >
                  Set up your first project's CI/CD
                </a>
              </div>
            </div>
          </div>
          <div
            class="gl-mb-4"
          >
            <div
              class="align-items-center flex"
            >
              <div>
                <a
                  class="gl-link"
                  data-testid="uncompleted-learn-gitlab-link"
                  data-track-action="click_link"
                  data-track-label="start_a_free_trial_of_gitlab_ultimate"
                  href="http://example.com/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Start a free trial of GitLab Ultimate
                </a>
              </div>
            </div>
          </div>
          <div
            class="gl-mb-4"
          >
            <div
              class="align-items-center flex"
            >
              <div>
                <a
                  class="gl-link"
                  data-testid="uncompleted-learn-gitlab-link"
                  data-track-action="click_link"
                  data-track-label="add_code_owners"
                  href="http://example.com/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Add code owners
                </a>
                <span
                  class="gl-font-style-italic gl-text-gray-500"
                  data-testid="trial-only"
                >
                  - Included in trial
                </span>
              </div>
            </div>
          </div>
          <div
            class="gl-mb-4"
          >
            <div
              class="align-items-center flex"
            >
              <div>
                <a
                  class="gl-link"
                  data-testid="uncompleted-learn-gitlab-link"
                  data-track-action="click_link"
                  data-track-label="enable_require_merge_approvals"
                  href="http://example.com/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Enable require merge approvals
                </a>
                <span
                  class="gl-font-style-italic gl-text-gray-500"
                  data-testid="trial-only"
                >
                  - Included in trial
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-lg-4 col-mb-6 col-sm-12 gl-mt-5"
    >
      <div
        class="gl-card gl-pt-0 h-100"
      >
        <div
          class="gl-bg-white gl-border-0 gl-card-header gl-pb-0"
        >
          <img
            src="deploy.svg"
          />
          <h2
            class="gl-font-lg gl-mb-3"
          >
            Deploy
          </h2>
          <p
            class="gl-mb-6 gl-text-gray-700"
          >
            Use your new GitLab workflow to deploy your application, monitor its health, and keep it secure:
          </p>
        </div>
        <div
          class="gl-card-body gl-pt-0"
        >
          <div
            class="gl-mb-4"
          >
            <div
              class="align-items-center flex"
            >
              <div>
                <a
                  class="gl-link"
                  data-testid="uncompleted-learn-gitlab-link"
                  data-track-action="click_link"
                  data-track-label="scan_dependencies_for_licenses"
                  href="https://docs.gitlab.com/ee/foobar/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Scan dependencies for licenses
                </a>
                <span
                  class="gl-font-style-italic gl-text-gray-500"
                  data-testid="trial-only"
                >
                  - Included in trial
                </span>
              </div>
            </div>
          </div>
          <div
            class="gl-mb-4"
          >
            <div
              class="align-items-center flex"
            >
              <div>
                <a
                  class="gl-link"
                  data-testid="uncompleted-learn-gitlab-link"
                  data-track-action="click_link"
                  data-track-label="scan_dependencies_for_vulnerabilities"
                  href="https://docs.gitlab.com/ee/foobar/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Scan dependencies for vulnerabilities
                </a>
                <span
                  class="gl-font-style-italic gl-text-gray-500"
                  data-testid="trial-only"
                >
                  - Included in trial
                </span>
              </div>
            </div>
          </div>
          <div
            class="gl-mb-4"
          >
            <div
              class="align-items-center flex"
            >
              <div>
                <a
                  class="gl-link"
                  data-testid="uncompleted-learn-gitlab-link"
                  data-track-action="click_link"
                  data-track-label="analyze_your_application_for_vulnerabilities_with_dast"
                  href="https://docs.gitlab.com/ee/foobar/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Analyze your application for vulnerabilities with DAST
                </a>
                <span
                  class="gl-font-style-italic gl-text-gray-500"
                  data-testid="trial-only"
                >
                  - Included in trial
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-lg-4 col-mb-6 col-sm-12 gl-mt-5"
    >
      <div
        class="gl-card gl-pt-0 h-100"
      >
        <div
          class="gl-bg-white gl-border-0 gl-card-header gl-pb-0"
        >
          <img
            src="plan.svg"
          />
          <h2
            class="gl-font-lg gl-mb-3"
          >
            Plan and execute
          </h2>
          <p
            class="gl-mb-6 gl-text-gray-700"
          >
            Create a workflow for your new workspace, and learn how GitLab features work together:
          </p>
        </div>
        <div
          class="gl-card-body gl-pt-0"
        >
          <div
            class="gl-mb-4"
          >
            <div
              class="align-items-center flex"
            >
              <div>
                <a
                  class="gl-link"
                  data-testid="uncompleted-learn-gitlab-link"
                  data-track-action="click_link"
                  data-track-label="create_an_issue"
                  href="http://example.com/"
                  target="_self"
                >
                  Create an issue
                </a>
              </div>
            </div>
          </div>
          <div
            class="gl-mb-4"
          >
            <div
              class="align-items-center flex"
            >
              <div>
                <a
                  class="gl-link"
                  data-testid="uncompleted-learn-gitlab-link"
                  data-track-action="click_link"
                  data-track-label="submit_a_merge_request_mr"
                  href="http://example.com/"
                  target="_self"
                >
                  Submit a merge request (MR)
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div />
</div>
`;
